cmd/gokeyless/gokeyless.go

```diff
@@ -93,41 +212,69 @@ func main() {
 			log.Info("already configured; exiting")
 		}
 		os.Exit(0)
+	case outputConfigMode:
+		b, err := yaml.Marshal(config)
+		if err != nil {
+			log.Fatal(err)
+		}
+		fmt.Print(string(b))
+		os.Exit(0)
 	}
 
 	// If we make it here we need to ask for user input, so we need to give up
 	// and log an error instead (in case the server is running as a daemon).
 	// Failing hard with an error message makes the problem obvious, whereas a
 	// daemon blocked waiting on input can be hard to debug.
 	if needNewCertAndKey() {
-		log.Error("the server cert/key need to be generated; run the server with either the -config-only or -manual-activation flag to generate the pair")
-		os.Exit(1)
+		if needInteractivePrompt() {
+			log.Error("the server cert/key need to be generated; set the hostname, zone_id, and origin_ca_key values in your config file, or run the server with either the -config-only or -manual-activation flag to generate the pair interactively")
+			os.Exit(1)
+		}
+		initializeServerCertAndKey()
 	}
 
 	cfg := server.DefaultServeConfig()
-	s, err := server.NewServerFromFile(cfg, certFile, keyFile, caFile)
+	s, err := server.NewServerFromFile(cfg, config.CertFile, config.KeyFile, config.CACertFile)
 	if err != nil {
 		log.Fatal("cannot start server:", err)
 	}
 
-	keys, err := server.NewKeystoreFromDir(keyDir, LoadKey)
+	keys, err := initKeyStore()
 	if err != nil {
 		log.Fatal(err)
 	}
-
 	s.SetKeystore(keys)
 
-	if pidFile != "" {
-		if f, err := os.Create(pidFile); err != nil {
-			log.Errorf("error creating pid file: %v", err)
+	if config.PidFile != "" {
+		if f, err := os.Create(config.PidFile); err != nil {
+			log.Fatalf("error creating pid file: %v", err)
 		} else {
 			fmt.Fprintf(f, "%d", os.Getpid())
 			f.Close()
 		}
 	}
 
-	go func() { log.Critical(s.MetricsListenAndServe(metricsAddr)) }()
-	log.Fatal(s.ListenAndServe(net.JoinHostPort("", port)))
+	go func() {
+		log.Critical(s.MetricsListenAndServe(net.JoinHostPort("", strconv.Itoa(config.MetricsPort))))
+	}()
+	log.Fatal(s.ListenAndServe(net.JoinHostPort("", strconv.Itoa(config.Port))))
```

Comment:
We imported strconv just for Itoa when we already have fmt?

Reply:
So we did. But the strconv package is already imported elsewhere in this project so there is no impact to binary size.